openmc/tallies.py

```diff
@@ -2522,6 +2522,68 @@ def __neg__(self):
         new_tally = self * -1
         return new_tally
 
+    def get_data_slice_where(self, basis: str='xy', slice_value: float=None):
+        """Gets a slice of tally scored on a RegularMesh to the mesh and gets a 2D slice of values.
+
+        Parameters
+        ----------
+        datasets : np.ndarray
+            1-D array of data values. Will be reshaped to fill the mesh and
+            must have the same number of entries to fill the mesh
+        basis : {'xy', 'xz', 'yz'}
+            The basis directions for the slice
+        slice_value : int
+            The axis value of the slice to extract.
+
+        Returns
+        -------
+        np.ndarray
+            the 2D array of dataset values
+        """
+
+        cv.check_value('basis', basis, ['xy', 'xz', 'yz'])
+        
+        mesh = self.find_filter(filter_type=openmc.MeshFilter).mesh
+        if isinstance(mesh, openmc.RegularMesh):
+            print('mesh accepted')
+
+        if len(self.scores) != 1:
```

Comment:
Let's add a `score` argument to this function to make it more flexible. There should probably be a `value` argument too à la `Tally.get_values`.

Reply:
Yep that sounds good. In the event that there is only one score I've coded it so the user doesn't need to specify the score